src/Grids/finitedifference.jl

```diff
@@ -45,9 +45,24 @@ mutable struct FiniteDifferenceGrid{
 end
 
 
-@memoize WeakValueDict function FiniteDifferenceGrid(
-    topology::Topologies.IntervalTopology,
-)
+# we can use a WeakValueDict to memoize here, as FiniteDifferenceGrids cannot be distributed, so won't suffer from
+#   https://github.com/CliMA/ClimaCore.jl/issues/1589
+const FD_GRID_CACHE = WeakValueDict()
```

Comment:
If this is the only place where we use `WeakValueDict`, would it make sense to use `Dict` instead? 
The two reasons I see to use a `Dict` are:
- We treat this grid in the same way as the other ones
- We remove a direct dependency from a package that hasn't been updated in 5 years, and might increase our surface area for problems in the future

Reply:
That's a good point.